idea/src/org/jetbrains/kotlin/idea/inspections/KotlinRedundantOverrideInspection.kt

```diff
@@ -77,6 +79,14 @@ class KotlinRedundantOverrideInspection : AbstractKotlinInspection(), CleanupLoc
         val superCallMethodName = superSelectorExpression.getCallNameExpression()?.text ?: return false
         return function.name == superCallMethodName
     }
+    
+    private fun KtNamedFunction.hasDerivedProperty(): Boolean {
+        if (valueParameters.isNotEmpty()) return false
+        val functionName = nameAsName ?: return false
+        if (!functionName.asString().startsWith("get")) return false
+        val propertyName = SyntheticJavaPropertyDescriptor.propertyNameByGetMethodName(functionName)?.asString() ?: return false
+        return containingClassOrObject?.body?.children?.any { it is KtProperty && it.name == propertyName } == true
```

Comment:
Use `containingClassOrObject?.declarations?.find { it is KtProperty && it.name == propertyName } != null`.